### PR TITLE
Mark inactive driver feature

### PR DESCRIPTION
**Pull request by goscha01**
My feature also merged with Jia's feature on the state of January the 7th, because I couldn't implement my feature without knowing the driver's credentials.

### CODE REVIEW

File: .sonarcloud.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (carolynrehm) [inline .sonarcloud.properties]: this file shouldn't be included in this pull request as there were no substantive changes

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -26,10 +26,10 @@ describe('AppComponent', () => {
     expect(app.title).toEqual('rideshare-frontend');
   });
 
-  it('should render title', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-    fixture.detectChanges();
-    const compiled = fixture.debugElement.nativeElement;
-    expect(compiled.querySelector('.content span').textContent).toContain('rideshare-frontend app is running!');
-  });
+  // it('should render title', () => {
```
Comment (carolynrehm) [inline src/app/app.component.spec.ts]: this commented out code should be removed

File: src/styles.css [hunk 3]
```diff
@@ -1 +1 @@
-/* You can add global styles to this file, and also import other style files */
+@import "~bootstrap/dist/css/bootstrap.min.css";
```
Comment (carolynrehm) [inline src/styles.css]: this should be configured in the angular.json file, not in the global styles

File: src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  //     observe: "response" as "body"
+  //   };
+
+
+
+
+  constructor(private httpClient: HttpClient) { }
+
+
+getDriverById(id: number): Observable <any>{
```
Comment (carolynrehm) [inline src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts]: Are the operations found in this file operations done on the user model? Marking a driver inactive is updating a user, correct? Why not include this functionality in the User service?

File: src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export class MarkInactiveDriverService {
+
+  // URL
+  private url = "http://localhost:8080/users/"
+  // private body: string;
+  // // Http options
+  //   private httpOptions = {
+  //     headers: new HttpHeaders({"Content-Type": "application/json"}),
+  //     observe: "response" as "body"
+  //   };
```
Comment (carolynrehm) [inline src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts]: remove commented code

File: src/app/components/driver/driver.component.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  templateUrl: './driver.component.html',
+  styleUrls: ['./driver.component.css']
+})
+export class DriverComponent implements OnInit {
+
+  userdriver : User ;
+
+  riders: User[];
+
+  userId = '';
```
Comment (carolynrehm) [inline src/app/components/driver/driver.component.ts]: Why use all of these instance variables rather than just using the instance variables of the user that you already have? Assign userdriver to an instance of user with these values. Also, keep in mind naming conventions for things like "userDriver" and "markInactiveDriverService"

**Review by carolynrehm — CHANGES_REQUESTED** [review]
A suggestion was made in Jia's pull request (#2) to handle authentication in the AuthService rather than in the sessionStorage. There are a few other suggestions on code that this pull request relies on that needs to be changed before accepted.  See comments for feedback on files that seem to be independent of that pull request

**Comment by goscha01** [comment]
The User service was developed by JIA, means it was a different feature and I didn’t have an access to it until the last time. Sent from Mail for Windows 10 From: carolynrehmSent: Thursday, January 9, 2020 11:46 AMTo: revaturelabs/rideshare-frontendCc: George Sayapin; AuthorSubject: Re: [revaturelabs/rideshare-frontend] Mark inactive driver feature (#6) @carolynrehm commented on this pull request.In src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts:> +  // URL+  private url = "http://localhost:8080/users/"+  // private body: string;+  // // Http options+  //   private httpOptions = {+  //     headers: new HttpHeaders({"Content-Type": "application/json"}),+  //     observe: "response" as "body"+  //   };+++++  constructor(private httpClient: HttpClient) { }+++getDriverById(id: number): Observable <any>{Are the operations found in this file operations done on the user model? Marking a driver inactive is updating a user, correct? Why not include this functionality in the User service?—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by goscha01** [comment]
If I understood your request correctly, the userDriver is an object and the riders is an array. So assign the first to an object and the second to an array. All parameters that were bellow these assignment I deleted. I put them when tried to get rid of errors in CircleCI.  Sent from Mail for Windows 10 From: carolynrehmSent: Thursday, January 9, 2020 11:49 AMTo: revaturelabs/rideshare-frontendCc: George Sayapin; AuthorSubject: Re: [revaturelabs/rideshare-frontend] Mark inactive driver feature (#6) @carolynrehm commented on this pull request.In src/app/components/driver/driver.component.ts:> +import { Batch } from 'src/app/models/batch';++++@Component({+  selector: 'app-driver',+  templateUrl: './driver.component.html',+  styleUrls: ['./driver.component.css']+})+export class DriverComponent implements OnInit {++  userdriver : User ;++  riders: User[];++  userId = ''; Why use all of these instance variables rather than just using the instance variables of the user that you already have? Assign userdriver to an instance of user with these values. Also, keep in mind naming conventions for things like "userDriver" and "markInactiveDriverService"—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

File: src/app/components/driver/driver.component.ts [hunk 7]
```diff
@@ -13,49 +13,42 @@ import { Batch } from 'src/app/models/batch';
...
+
+
+
    token = parseInt(sessionStorage.getItem("auth"));
    
    
    
    
-  constructor(private _MarkInactiveDriverService_: MarkInactiveDriverService,  private router: Router) { }
+  constructor(private _markInactiveDriverService_: MarkInactiveDriverService,  private router: Router) { }
```
Comment (carolynrehm) [inline src/app/components/driver/driver.component.ts]: why the underscores here?

File: src/app/components/navbar/navbar.component.ts [hunk 8]
```diff
@@ -48,10 +48,10 @@ export class NavbarComponent implements OnInit {
...
     });
 
-    this.userService.getEmitter().subscribe((user: User) => {
-      this.token = user.userId;
-      this.name = user.firstName;
-    });
+    // this.userService.getEmitter().subscribe((user: User) => {
+    //   this.token = user.userId;
+    //   this.name = user.firstName;
+    // });
```
Comment (carolynrehm) [inline src/app/components/navbar/navbar.component.ts]: why is this commented out?

File: src/app/components/driver/driver.component.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+       if(userdriver.acceptingRides == true){
+        userdriver.acceptingRides = false;
+      this.markInactiveDriverService.changeDriverIsAccepting (this.userDriver);
+      
+    }
+    else {
+      console.log ("driver else before", this.userDriver);
+      userdriver.acceptingRides = true;
+      this.markInactiveDriverService.changeDriverIsAccepting(this.userDriver);
+      console.log ("driver else after", this.userDriver);
```
Comment (carolynrehm) [inline src/app/components/driver/driver.component.ts]: create a logging service or remove console.logs

Comment (carolynrehm) [reply src/app/components/driver/driver.component.ts]: -- this can be added to a new feature --

File: src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts (hunk 4 above)
Comment (carolynrehm) [reply src/app/services/mark-inactive-driver-service/mark-inactive-driver.service.ts]: @goscha01 can we integrate this into the user service?

File: .sonarcloud.properties (hunk 1 above)
Comment (carolynrehm) [reply .sonarcloud.properties]: the sonarcloud properties should not be removed all together, they just shouldn't be any different from dev

**Review by carolynrehm — CHANGES_REQUESTED** [review]
@goscha01 changes look good. Just two more things to take care of:

(1) Migrating the inactive driver service implementation into the user service
(2) Returning the sonar-project.properties (we don't want to delete it in the PR, but rather have it not included in the pull request at all - indicating that there were no changes from the dev branch to your feature branch in this file)

Once those are complete, we should be all set to merge!

**Review by carolynrehm — CHANGES_REQUESTED** [review]
"sonar-cloud.properties" --> this file's name was changed (name is important for sonar) so we'll need it changed back to sonar-project.properties. Then we should be all set @goscha01

**Review by carolynrehm — APPROVED** [review]

**Comment by carolynrehm** [comment]
@goscha01 this has been approved and merged, but there are some residual bugs and code smells. Please take a look at them on sonarcloud and resolve them
